software/wmla120.py

```diff
@@ -1385,17 +1386,40 @@ def _load_filelist(self):
                 self.globs = file_lists['globs']
                 self.files = file_lists['files']
 
-    def _add_dependent_packages(self, repo_dir, dep_list):
-        cmd = (f'yumdownloader --archlist={self.arch} --destdir '
-               f'{repo_dir} {dep_list}')
-        resp, err, rc = sub_proc_exec(cmd)
-        if rc != 0:
-            self.log.error('An error occurred while downloading dependent packages\n'
-                           f'rc: {rc} err: {err}')
-        resp = resp.splitlines()
-        for item in resp:
-            if 'No Match' in item:
-                self.log.error(f'Dependent packages download error. {item}')
+    def _add_dependent_packages(self, repo_dir, dep_list, also_get_newer=True):
+        if also_get_newer:
+            dep_list_list = dep_list.split()
+            versionless_dep_list, ver = parse_rpm_filenames(dep_list_list)
+            versionless_dep_list = ' '.join(versionless_dep_list)
+
+            cmd = (f'yumdownloader --archlist={self.arch} --destdir '
+                   f'{repo_dir} {versionless_dep_list}')
+            resp, err, rc = sub_proc_exec(cmd)
+            if rc != 0:
+                self.log.error('An error occurred while downloading dependent packages\n'
+                               f'rc: {rc} err: {err}')
+            resp = resp.splitlines()
```

Comment:
can you also add the cmd to the output in the error log?  also seems like you are replicating code ... 1395-1401 is the same as 1413-1419 ... simple method can improve usability of the code.

Reply:
done